flink-libraries/flink-gelly/src/test/java/org/apache/flink/graph/library/similarity/JaccardIndexTest.java

```diff
@@ -0,0 +1,136 @@
+import org.apache.flink.test.util.TestBaseUtils;
+import org.apache.flink.types.IntValue;
+import org.apache.flink.types.LongValue;
+import org.apache.flink.types.NullValue;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class JaccardIndexTest
+extends AsmTestBase {
+
+	@Test
+	public void testSimpleGraph()
+			throws Exception {
+		DataSet<Result<IntValue>> ji = undirectedSimpleGraph
+			.run(new JaccardIndex<IntValue, NullValue, NullValue>());
+
+		String expectedResult =
+			"(0,1,(1,4))\n" +
+			"(0,2,(1,4))\n" +
+			"(0,3,(2,4))\n" +
+			"(1,2,(2,4))\n" +
+			"(1,3,(1,6))\n" +
+			"(1,4,(1,3))\n" +
+			"(1,5,(1,3))\n" +
+			"(2,3,(1,6))\n" +
+			"(2,4,(1,3))\n" +
+			"(2,5,(1,3))\n" +
+			"(4,5,(1,1))\n";
+
+		TestBaseUtils.compareResultAsText(ji.collect(), expectedResult);
+	}
+
+	@Test
+	public void testSimpleGraphWithMinimumScore()
+			throws Exception {
+		DataSet<Result<IntValue>> ji = undirectedSimpleGraph
+			.run(new JaccardIndex<IntValue, NullValue, NullValue>()
+				.setMinimumScore(1, 2));
+
+		String expectedResult =
+			"(0,3,(2,4))\n" +
+			"(1,2,(2,4))\n" +
+			"(4,5,(1,1))\n";
+
+		TestBaseUtils.compareResultAsText(ji.collect(), expectedResult);
+	}
+
+	@Test
+	public void testSimpleGraphWithMaximumScore()
+			throws Exception {
+		DataSet<Result<IntValue>> ji = undirectedSimpleGraph
+			.run(new JaccardIndex<IntValue, NullValue, NullValue>()
+				.setMaximumScore(1, 2));
+
+		String expectedResult =
+			"(0,1,(1,4))\n" +
+			"(0,2,(1,4))\n" +
+			"(1,3,(1,6))\n" +
+			"(1,4,(1,3))\n" +
+			"(1,5,(1,3))\n" +
+			"(2,3,(1,6))\n" +
+			"(2,4,(1,3))\n" +
+			"(2,5,(1,3))\n";
+
+		TestBaseUtils.compareResultAsText(ji.collect(), expectedResult);
+	}
+
+	@Test
+	public void testCompleteGraph()
+			throws Exception {
+		DataSet<Result<LongValue>> ji = completeGraph
+			.run(new JaccardIndex<LongValue, NullValue, NullValue>()
+				.setGroupSize(4));
+
+		for (Result<LongValue> result : ji.collect()) {
+			// the intersection includes every vertex
+			assertEquals(completeGraphVertexCount, result.getDistinctNeighborCount().getValue());
+
+			// the union only excludes the two vertices from the similarity score
+			assertEquals(completeGraphVertexCount - 2, result.getCommonNeighborCount().getValue());
+		}
+	}
+
+	@Test
+	public void testRMatGraph()
+			throws Exception {
+		long vertexCount = 1 << 8;
+		long edgeCount = 8 * vertexCount;
+
+		RandomGenerableFactory<JDKRandomGenerator> rnd = new JDKRandomGeneratorFactory();
+
+		Graph<LongValue, NullValue, NullValue> graph = new RMatGraph<>(env, rnd, vertexCount, edgeCount)
+			.setSimpleGraph(true, false)
+			.generate();
+
+		DataSet<Result<LongValue>> ji = graph
+			.run(new JaccardIndex<LongValue, NullValue, NullValue>()
+				.setGroupSize(4));
+
+		ChecksumHashCode checksum = DataSetUtils.checksumHashCode(ji);
+
+		assertEquals(13954, checksum.getCount());
+		assertEquals(0x0000179f83a2a873L, checksum.getChecksum());
```

Comment:
These are regression tests and cover aspects of the algorithms not tested by hand-crafted inputs.